packages/@react-aria/overlays/src/usePreventScroll.ts

```diff
@@ -59,7 +65,8 @@ export function usePreventScroll(options: PreventScrollOptions = {}) {
 function preventScrollStandard() {
   return chain(
     setStyle(document.documentElement, 'paddingRight', `${window.innerWidth - document.documentElement.clientWidth}px`),
-    setStyle(document.documentElement, 'overflow', 'hidden')
+    setStyle(document.documentElement, 'overflow', 'hidden'),
+    () => preventScrollCount--
```

Comment:
actually, rather than having this sprinkled everywhere, you could move it like this
```
    preventScrollCount++;
    if (isIOS()) {
      return chain(() => preventScrollCount--, preventScrollMobileSafari());
    } else {
      return chain(() => preventScrollCount--, preventScrollStandard());
    }
```
then everything in those functions and returned functions can be agnostic of adding/removing, they just check for 0 or > 0

does that make sense?